### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by ich777**
Please test, this should look something like that:
grafik

@mtongnz what do you think about these changes?

### CODE REVIEW

**Comment by ich777** [comment]
What about hiding the IPs and ports entirely when the container is not running like:
grafik

**Comment by mrsdoubtfire613** [comment]
@ich777  Host, macvlan, and bridge work.  I do like the idea of hiding IPs and Ports when not running as you showed.
image

Only containers that don't work are those created as compose stacks:
1. In the screenshot above, I have the single container compose stack (pluginMakemkvTest) connected to the macvlan bond0.43 *and* the default-created bridge network.  As you can see, no external IP:Port combos appear at all.  Interestingly, the webUI URL is generated and links to the bond0.43 IP:PORT ("10.43.64.34:6080") respecting the container port number and ignoring the reassigned host port (the compose file maps 6090:6080).
2. In the next screenshot, I removed the bond0.43 network and set only the compose-created "default" network.  The external IP:PORT appear.  However, the webui URL doesn't generate, the "link" is exactly as written in the compose file "http://[IP]:[PORT:6080]....".
image
3. In the following screenshot, I've set the compose container to only use the bond0.43 network.  Again, no external IP:PORT is displayed *but* the webui URL is properly generated.
image


I am also still having the strange interaction with the basic view/advanced view toggle.  I'll try to run in safe mode after work today to see if there is a plugin causing an issue.

**Comment by ich777** [comment]
@mrsdoubtfire613 the strange behaviour is maybe caused by the Folder View plugin since I can't reproduce the behaviour you've described over here. IIRC correctly the plugin modifies the Docker page and because of these two new rows or better speaking different rows it could be the case that you see such behaviour.

> Only containers that don't work are those created as compose stacks:

I don't have the Compose plugin installed either. I think the first thing that needs to be working is the base functionality without any 3rd party plugins and then focusing on the other things.
I'll have to go through your compose yaml and see how that creates the network.

**Comment by mrsdoubtfire613** [comment]
> @mrsdoubtfire613 the strange behaviour is maybe caused by the Folder View plugin since I can't reproduce the behaviour you've described over here. IIRC correctly the plugin modifies the Docker page and because of these two new rows or better speaking different rows it could be the case that you see such behaviour.

I suspect that plugin as well.  I'll report back later today after I can disable it and retest.

> I don't have the Compose plugin installed either. I think the first thing that needs to be working is the base functionality without any 3rd party plugins and then focusing on the other things. I'll have to go through your compose yaml and see how that creates the network.

Makes sense.  I'll hold off any comments on compose-created containers until such a time as the base functionality improvement is made and until you let me know you'd like to look into the compose functionality.

**Comment by ich777** [comment]
@mrsdoubtfire613 no worries, I just want to get the basics working and this PR into unRAID itself, after that plugin Authors can look why or what is not working if they inject/change code into the base when installing the plugin.

**Comment by ich777** [comment]
@mrsdoubtfire613 I found a few bugs, can you please retest with the newest fixes?

**Comment by mrsdoubtfire613** [comment]
@ich777 I rebooted my server into safe mode to test.  I manually updated the DockerContainers.page, /include/DockerContainers.php, and /include/DockerClient.php files with your PR.  All of the testing below is in safe mode with only the 3 nginx dockers running (no plugins, no VMs, no other containers) and having modified only the 3 files I just listed:


**Containers and WebUI URLs**
image
"Host" and "MacVLAN" network types show the correct external IP:PORT pairing and the WebUI URL generated is correct. Note - the LSIO NGINX container doesn't work in 'host' mode as it tries to bind port 80 and 443 so selecting the webUI URL just brings up the unraid login page.

"Bridge" network type containers show the correct External IP:PORT pairing *but* the webUI URL does *not* get generated, it appears as 'http://[IP]:[PORT:80]' (or whatever is defined as the webUI).  My apologies as I may have missed this previously when testing when I said that bridge type containers were working fine as I haven't made any changes to the containers since my previous statement except for rebooting into safe mode.  I changed another random container (created through the unraid GUI, not compose, and not one I've used in testing before) to bridge mode and experienced the same incorrect/ungenerated webUI URL.  If I manually point my browser to the appropriate IP:PORT combo for the webUI, the page loads, so at least I know the container is actually running and binding properly.


**Basic/Advanced Toggle**
tl;dr: I'm still having the basic/advanced toggle issue.  No matter the switch position, loading the Docker tab shows advanced view and the toggle shows whatever the previous toggle state was independent of the actual page view.

Longer:
The first load of the docker tab shows "basic view" but is advanced.  If I select the toggle, the page changes to basic view while the toggle shows "advanced view".  If - while the page shows basic and toggle shows advanced - I then refresh the page, the toggle shows "advanced view" and the page is advanced.  If I toggle back to basic and refresh the page, the page shows advanced and the toggle is "basic".

**Comment by ich777** [comment]
> "Bridge" network type containers show the correct External IP:PORT pairing _but_ the webUI URL does _not_ get generated, it appears as 'http://[IP]:[PORT:80]' (or whatever is defined as the webUI). My apologies as I may have missed this previously when testing when I said that bridge type containers were working fine as I haven't made any changes to the containers since my previous statement except for rebooting into safe mode. I changed another random container (created through the unraid GUI, not compose, and not one I've used in testing before) to bridge mode and experienced the same incorrect/ungenerated webUI URL. If I manually point my browser to the appropriate IP:PORT combo for the webUI, the page loads, so at least I know the container is actually running and binding properly.

This was caused due to a bug that I just found, this should now work with the newest changes (only `DockerClient.php` was changed).
Maybe the Compose issue is now also partially solved?


> **Basic/Advanced Toggle**

I still can't reproduce this behaviour, can you try to clear your cache or at least try a Private Browser window and see if that fixes the issue? I think it is maybe related to something stored in the cache. You can also try another Browser that you usually don't use.
Here is what it looks like on my server (I haven't reload my page here but the initial view is also without Advanced View enabled):
gif

**Comment by mrsdoubtfire613** [comment]
> This was caused due to a bug that I just found, this should now work with the newest changes (only `DockerClient.php` was changed). Maybe the Compose issue is now also partially solved?

Tested with new changes and webui URLs work now, thank you.

> I still can't reproduce this behaviour, can you try to clear your cache or at least try a Private Browser window and see if that fixes the issue? I think it is maybe related to something stored in the cache. You can also try another Browser that you usually don't use. Here is what it looks like on my server (I haven't reload my page here but the initial view is also without Advanced View enabled):

Sure.  I recorded two screencaps - one from a Chrome incognito browser tab and one from a Firefox Private Browsing tab - and have placed them below.  I also wiped the cache of each browser and ran the same check in both browsers with the same result, I just didn't screencap those attempts.  In both instances I open or have just opened the docker tab and start using the toggle.  I then refresh the page (Ctrl + r) to show the effects of reloading, change the toggle, and refresh again to demonstrate.

*Firefox Private Browsing Tab*
https://github.com/user-attachments/assets/1ff59de4-0648-4f6d-9daa-69cf7f24b535

*Chrome Incognito Tab*
https://github.com/user-attachments/assets/8336714d-e51b-480f-9552-924d92eb23b2

Update:  Those screencaps were taken while the server is still in safe mode.

**Comment by ich777** [comment]
> Update: Those screencaps were taken while the server is still in safe mode.

Something really weird is happening on your system and I still think it's caused through caching:
- The move indicators shouldn be even visible when the lock is closed
- The `OFF` text should be right next to the switch not below it
grafik

Here is a screencap from what it looks like on my system when I open up the Docker tab (just opened the browser and just typed in the URL):

peek

**Comment by ich777** [comment]
@mtongnz what do you think about that change?

**Comment by mrsdoubtfire613** [comment]
> Something really weird is happening on your system and I still think it's caused through caching:

Are you referring to browser caching or some other caching?  If other, is there something else I should wipe/flush before testing again?

Were there any changes to the UI between 6.12.11 and the 7.0.0 betas because I see we're on different versions of unraid.

**Comment by ich777** [comment]
> Were there any changes to the UI between 6.12.11 and the 7.0.0 betas because I see we're on different versions of unraid.

The main/master branch is for the newest version from Unraid, so to speak 7.0.0
However it should work on 6.12.11 too.

**Comment by mrsdoubtfire613** [comment]
Alright.  I'm not sure how else to troubleshoot save for updating to version 7 and testing to confirm so for now I'll chock it up to something weird on my end and hope someone else can test to confirm it works fully as it does on your machine or confirm the odd UI behavior I'm seeing.

Only comment to add because you mentioned compose before is that for a compose stack attached to multiple networks there are still no external IPs displayed at all.  However, the WebUI url is generated properly for the connection to the bond0.43 network.
image

Thanks for putting in the time for this, I think it's going to be a great upgrade to the UI once it's in place.

**Comment by ich777** [comment]
> Thanks for putting in the time for this, I think it's going to be a great upgrade to the UI once it's in place.

The main credits go to @mtongnz I only made a few modifications, I hope he sees this soon so that he can merge this PR so that it can be implemented in the next beta for further testing.

**Comment by mtongnz** [comment]
Sorry for the delay - work seems to always get in the way of my projects :)

@ich777 That looks great.  Thanks for the assist.

**Comment by mrsdoubtfire613** [comment]
@ich777 I see the PR was merged but wanted to update you: I updated to 7.0.0.0-beta2 and incorporated the full changes from your PR and @mtongnz changes and I no longer have any of the weird behavior with advanced view/basic view, unaligned on/off text/toggles, and the reorder icon showing when not activated.

**Comment by ich777** [comment]
@mrsdoubtfire613 good news, glad to hear that everything is working now. :)
